src/internet_identity/src/main.rs

```diff
@@ -31,6 +31,9 @@ const DEFAULT_SIGNATURE_EXPIRATION_PERIOD_NS: u64 = secs_to_nanos(600);
 // 5 mins
 const POW_NONCE_LIFETIME: u64 = secs_to_nanos(300);
 
+/// The maximum number of users we can store in a single canister.
+const MAX_USERS_PER_CANISTER: u64 = 2_000_000_000;
```

Comment:
I wonder if we should calculate this value (based on record size and memory size).

Or, maybe better, in `storage.set_user_number_range` and `Storage::new` trap if we’d need more storage than is available in stable memory (i.e. if the last byte of the the last record is out of bounds).

Or, in any case, something like
```suggestion
// This number is derived from the stable memory size (4GB) and the user record size (2kb)
// leaving a few pages in the end as spares
const MAX_USERS_PER_CANISTER: u64 = 2_000_000_000;
```

Reply:
Very good point. I changed the code to compute the max number of users based on the canister capacity and realized that the upper bound I specified is 1000x higher than it should be (it should be 2M, not 2B, *facepalm*).
The computed upper bound is `10_000 + 2_097_151`.